pkg/controller/storagecluster/external_resources_test.go

```diff
@@ -349,3 +355,34 @@ func assertCephObjectStore(t *testing.T, reconciler ReconcileStorageCluster, rem
 		assert.Equal(t, hostFound, cObjS.Spec.Gateway.ExternalRgwEndpoints[0].IP)
 	}
 }
+
+func startServerAt(endpoint string) <-chan error {
+	var doneChan = make(chan error)
+	go func(doneChan chan<- error) {
+		defer close(doneChan)
+		ln, err := net.Listen("tcp4", endpoint)
+		if err != nil {
+			doneChan <- err
+			return
+		}
+		defer ln.Close()
+		conn, err := ln.Accept()
+		if err != nil {
+			doneChan <- err
+			return
+		}
+		defer conn.Close()
+		doneChan <- nil
+	}(doneChan)
+	return doneChan
+}
+
+func generateRandomPort(minPort, maxPort int) int {
+	rand.Seed(time.Now().UnixNano())
+	portRange := minPort - maxPort
+	if portRange < 0 {
+		portRange *= -1
+	}
+	retPort := rand.Intn(portRange) + minPort
+	return retPort
+}
```

Comment:
Move these above the Test function

Reply:
Moved it along side with other helper functions in the file...